README.md

```diff
@@ -10,17 +10,43 @@
 [Examples](https://r-nimble.org/examples) |
 [Developing](https://nimble-dev.github.io/nimble-docs)
 
-NIMBLE is an R package for programming with BUGS models and compiling parts of R.
-[BUGS](https://www.mrc-bsu.cam.ac.uk/software/bugs) is a probabilistic programming language that makes it easy to rapidly develop hierarchical Bayesian models for statistical analysis.
-NIMBLE provides an implementation of the BUGS language in R together with a DSL for writing custom inference algorithms against BUGS models.
-NIMBLE's programming paradigm treats probabilistic graphical models as a basic programming construct.
+NIMBLE is an R package for hierarchical statistical modeling (aka
+graphical modeling).  It enables writing general models along with
+methods such as Markov chain Monte Carlo (MCMC), particle filtering
+(aka sequential Monte Carlo), and other general methods.
+
+For writing statistical models, NIMBLE adopts and extends the BUGS
+language, making it largely compatible with
+[BUGS](https://www.mrc-bsu.cam.ac.uk/software/bugs) and
+[JAGS](http://mcmc-jags.sourceforge.net/).  NIMBLE makes BUGS
+extensible, allowing users to add new functions and new distributions.
+
+For writing algorithms (aka analysis methods), NIMBLE provides a
+model-generic programming system, based on two-stage evaluation,
```

Comment:
It's a term from computer science.  But I removed it.  If it doesn't make sense to you, it is certainly too jargony.